tests/sentry/rules/processing/test_delayed_processing.py

```diff
@@ -1,53 +1,292 @@
+import copy
+from datetime import UTC, datetime
 from unittest.mock import Mock, patch
+from uuid import uuid4
+
+import pytest
 
 from sentry.db import models
+from sentry.eventstore.models import Event
 from sentry.rules.processing.delayed_processing import (
     apply_delayed,
     process_delayed_alert_conditions,
 )
-from sentry.testutils.cases import TestCase
+from sentry.testutils.cases import APITestCase, TestCase
+from sentry.testutils.factories import DEFAULT_EVENT_DATA
+from sentry.testutils.helpers.datetime import iso_format
+from tests.snuba.rules.conditions.test_event_frequency import BaseEventFrequencyPercentTest
 
+pytestmark = pytest.mark.sentry_metrics
 
-class ProcessDelayedAlertConditionsTest(TestCase):
-    def get_rulegroup_event_mapping_from_input(
-        self, model: type[models.Model], field: dict[str, models.Model | str | int]
-    ):
-        # There will only be one event per rulegroup
-        proj_id = field.popitem()[1]
-        return self.buffer_mapping[proj_id]
 
-    @patch("sentry.rules.processing.delayed_processing.safe_execute")
-    def test_fetches_from_buffer_and_executes(self, mock_safe_execute):
-        project_two = self.create_project()
+class ProcessDelayedAlertConditionsTest(TestCase, APITestCase, BaseEventFrequencyPercentTest):
+    def create_event(self, project_id, timestamp, fingerprint, environment=None) -> Event:
+        data = {
+            "timestamp": iso_format(timestamp),
+            "stacktrace": copy.deepcopy(DEFAULT_EVENT_DATA["stacktrace"]),
+            "environment": environment,
+            "fingerprint": [fingerprint],
+            "level": "error",
+            "user": {"id": uuid4().hex},
+            "exception": {
+                "values": [
+                    {
+                        "type": "IntegrationError",
+                        "value": "Identity not found.",
+                    }
+                ]
+            },
+        }
+        return self.store_event(
+            data=data,
+            project_id=project_id,
+            assert_no_errors=False,
+        )
+
+    def setUp(self):
+        super().setUp()
+        self.event_frequency_condition = {
+            "interval": "1d",
+            "id": "sentry.rules.conditions.event_frequency.EventFrequencyCondition",
+            "value": 1,
+            "name": "The issue is seen more than 1 times in 1d",
+        }
```

Comment:
nit: I'd recommend pulling this out as a function with kwargs for each key/value and then set default values for each. That would allow you to remove anything that's replicated between each event condition.
```python
def create_event_frequency_condition(
    interval="1d",
    id="sentry.rules.conditions.event_frequency.EventFrequencyCondition",
    value=1,
    name="The issue is seen more than 1 times in 1d",
):
    return {"interval": interval, "id": id, "value": value, "name": name}
```
then these can be simplified a little..
```python
self.event_frequency_condition = create_event_frequency_condition()
self.event_frequency_condition2 = create_event_frequency_condition(
    value=2, name="The issue is seen more than 2 times in 1d"
)
```
If the name is always tied to the other information, we could use an fstring and generate it instead.